Automattic/wp-calypso
# Jetpack Error UX: Resolve concerns about state sync

Fixes: https://github.com/Automattic/dotcom-forge/issues/4647

## Proposed Changes

This pr aims to decouple react query from our current state management.

cc: @flootr , @tyxla 

## Testing Instructions
1. Create a new WoA site.
2. Enable ssh.
3. Login to the site and add the following to your `wp-config`
```
if ( ! empty( $_GET['rest_route'] ) ) {
	http_response_code( 404 );
	die('not found');
}
```
4. Open developer console.
5. Visit eg `/posts` page
6. Ensure that you see an error.
7. Ensure that you see a request at `/test-connection`
8. Ensure that you see a request at `/jetpack-connection-health`
9. Reload the page
10. Ensure that `/jetpack-connection-health` endpoint is not called
11. Wait 5 minutes / or clear your IndexedDB cache
12. Ensure that `/jetpack-connection-health` endpoint is called again

Below is a video that shows the request caching in case of Jetpack health error:

https://github.com/Automattic/wp-calypso/assets/497103/9eb395fe-aac0-4d61-9241-38dfd2ab80a1





## Pre-merge Checklist
